### PR TITLE
vscode-extensions.charliermarsh.ruff: 2023.60.0 -> 2024.4.0

### PR DESCRIPTION
**Pull request by DataHearth**
## Description of changes

I updated the version to latest (2024.4.0) and added arch parameters

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Azd325 — COMMENTED** [review]
Can we rename the commit to
vscode-extensions.charliermarsh.ruff: 2023.60.0 -> 2024.4.0

Rest LGTM and did build locally

**Comment by DataHearth** [comment]
> Can we rename the commit to
> vscode-extensions.charliermarsh.ruff: 2023.60.0 -> 2024.4.0
> 
> 

Done!

**Comment by Azd325** [comment]
Hi! To fit [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), could you please rename the commit to `vscode-extensions.charliermarsh.ruff: 2023.60.0 -> 2024.4.0`? Thanks! :)

[pkgs commit-conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions)

**Comment by DataHearth** [comment]
> Hi! To fit [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), could you please rename the commit to `vscode-extensions.charliermarsh.ruff: 2023.60.0 -> 2024.4.0`? Thanks! :)
> 
> [pkgs commit-conventions](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#commit-conventions)

@Azd325 Oh my bad, sorry for the wasted time, I'll keep an eye on that for the next PR! Should be OK now. I also added explanation in commit description :+1:

**Review by Azd325 — APPROVED** [review]

**Comment by Azd325** [comment]
@DataHearth But nice catch. I have only darwin so I never saw that issue. :)

**Review by Azd325 — APPROVED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -753,18 +753,36 @@ let
...
+            };
+          };
+        in {
           name = "ruff";
           publisher = "charliermarsh";
-          version = "2023.60.0";
-          sha256 = "sha256-qgwud2gzHLHID45VxDlngFMoks5O3pTHQe+Q7bdf8+I=";
-        };
+          version = "2024.4.0";
+        } // sources.${stdenv.system};
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:779]:
```suggestion
        } // sources.${stdenv.system} or throw "Unsupported system ${stdenv.system}";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 2]
```diff
@@ -753,18 +753,36 @@ let
...
+              arch = "linux-x64";
+              sha256 = "d9cd2d1ff3250cea9ec9f7dc57c642cb8c28a2005371fd460ae3cf3bc2576967";
+            };
+            "x86_64-darwin" = {
+              arch = "darwin-x64";
+              sha256 = "7aebc62253bbf77d4de7a4790562aedc5f67484a0381ff835e4ec7825d6a494c";
+            };
+            "aarch64-linux" = {
+              arch = "linux-arm64";
+              sha256 = "1j1xlvbg3nrfmdd9zm6kywwicdwdkrq0si86lcndaii8m7sj5pfp";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:768]:
```suggestion
              sha256 = "sha256-dGpIHChnfrQbxRZDuoAi4imgStyyPdxdvTQ3lknMYu0=";
```

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 3]
```diff
@@ -753,18 +753,36 @@ let
...
+            };
+            "aarch64-darwin" = {
+              arch = "darwin-arm64";
+              sha256 = "746a481c28677eb41bc51643ba8022e229a04adcb23ddc5dbd34379649cc62ed";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:772]:
```suggestion
              sha256 = "sha256-tElX4C0I5AmpxSHMtqOsxSAUImD1tqArB5fnvhw4LFc=";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 4]
```diff
@@ -787,18 +787,36 @@ let
...
+              arch = "linux-x64";
+              sha256 = "sha256-2c0tH/MlDOqeyffcV8ZCy4woogBTcf1GCuPPO8JXaWc=";
+            };
+            "x86_64-darwin" = {
+              arch = "darwin-x64";
+              sha256 = "sha256-euvGIlO7931N56R5BWKu3F9nSEoDgf+DXk7Hgl1qSUw=";
+            };
+            "aarch64-linux" = {
+              arch = "linux-arm64";
+              sha256 = "sha256-190i9akoRtUsowZFDXCejTcWOffT1J9aqy7b8damPcg=";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:802]:
```suggestion
              sha256 = "sha256-dGpIHChnfrQbxRZDuoAi4imgStyyPdxdvTQ3lknMYu0=";
```
I think you swapped the hashes

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 5]
```diff
@@ -787,18 +787,36 @@ let
...
+            };
+            "aarch64-darwin" = {
+              arch = "darwin-arm64";
+              sha256 = "sha256-dGpIHChnfrQbxRZDuoAi4imgStyyPdxdvTQ3lknMYu0=";
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/vscode/extensions/default.nix:806]:
```suggestion
              sha256 = "sha256-tElX4C0I5AmpxSHMtqOsxSAUImD1tqArB5fnvhw4LFc=";
```

File: pkgs/applications/editors/vscode/extensions/default.nix (hunk 4 above)
Comment (DataHearth) [reply pkgs/applications/editors/vscode/extensions/default.nix]: @SuperSandro2000 You were right. I don't know with which file I mixed it up, but clearly not the good one... Sry for that

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**